packs/nagios/etc/st2service_handler.py

```diff
@@ -43,152 +48,306 @@
     try:
-        # sys.stdout.write('POST: url: %s, body: %s\n' % (url, body))
-        r = requests.post(url, data=json.dumps(body), headers=headers)
+        if verbose:
+            print('Webhook POST: url: {0}, headers: {1}, body: {2}\n'
+                  .format(url, headers, body))
+        r = requests.post(url, data=json.dumps(body), headers=headers,
+                          verify=False)
     except:
-        sys.stderr.write('Cannot connect to st2 endpoint.')
+        traceback.print_exc(20)
+        raise Exception('Cannot connect to st2 endpoint {0}.'.format(url))
     else:
         status = r.status_code
+
+        if status in UNREACHABLE_CODES:
+            msg = 'Webhook URL {0} does not exist. Check StackStorm '\
+                'installation!'.format(url)
+            raise Exception(msg)
+
         if status not in OK_CODES:
-            sys.stderr.write('Failed posting nagios event to st2. HTTP_CODE: %d\n' % status)
+            sys.stderr.write('Failed posting nagio event to st2. HTTP_CODE: '
+                             '{0}\n'.format(status))
         else:
-            sys.stdout.write('Sent nagios event to st2. HTTP_CODE: %d\n' % status)
+            sys.stdout.write('Sent nagio event to st2. HTTP_CODE: '
+                             '{0}\n'.format(status))
+
+
+def _post_event_to_st2(payload, verbose=False):
+    body = {}
+    body['trigger'] = ST2_TRIGGERTYPE_REF
+    try:
+        body['payload'] = payload
+    except:
+        traceback.print_exc(limit=20)
+        print('Invalid JSON payload {0}.'.format(payload))
+        sys.exit(3)
+
+    try:
+        _post_webhook(url=_get_st2_webhooks_url(), body=body, verbose=verbose)
+        return True
+    except:
+        traceback.print_exc(limit=20)
+        print('Cannot send event to st2.')
+        sys.exit(4)
 
 
-def _get_payload(host, service, event_id, state, state_type, attempt):
+def _set_config_opts(config_file, verbose=False):
+    global ST2_USERNAME
+    global ST2_PASSWORD
+    global ST2_API_KEY
+    global ST2_AUTH_TOKEN
+    global ST2_API_BASE_URL
+    global ST2_AUTH_BASE_URL
+    global ST2_SSL_VERIFY
+    global UNAUTHED
+    global IS_API_KEY_AUTH
+    if not os.path.exists(config_file):
+        print('Configuration file "{0}" not found. Exiting!!!'
+              .format(config_file))
+        sys.exit(1)
+
+    with open(config_file) as f:
+        config = yaml.safe_load(f)
+
+        if verbose:
+            print('Contents of config file: {0}'.format(config))
+
+        ST2_USERNAME = config['st2_username']
+        ST2_PASSWORD = config['st2_password']
+        ST2_API_KEY = config.get('st2_api_key', None)
+        ST2_API_BASE_URL = config['st2_api_base_url']
+        if not ST2_API_BASE_URL.endswith('/'):
+            ST2_API_BASE_URL += '/'
+        ST2_AUTH_BASE_URL = config['st2_auth_base_url']
+        if not ST2_AUTH_BASE_URL.endswith('/'):
+            ST2_AUTH_BASE_URL += '/'
+        if type(config['unauthed']) is bool:
```

Comment:
Why are you doing a type check? In the sample config, set boolean values and if people use strings, it's their problem.